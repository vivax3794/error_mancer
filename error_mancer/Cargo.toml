[package]
name = "error_mancer"
version = "0.2.2"
edition = "2021"
description = "Master your errors with precision, context, and effortless control."
license = "MIT"

[dependencies]
error_mancer_macros = {path = "../error_mancer_macros", version="0.2.1"}

[dev-dependencies]
trybuild = "1"
anyhow = "1"
thiserror = "2"
