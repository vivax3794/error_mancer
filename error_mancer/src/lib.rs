pub mod prelude {
    pub use error_mancer_macros::errors;

    pub use super::handle;
}

#[macro_export]
macro_rules! handle {
    {$expr:expr => $error:path {propagate ($($return_error:ident),*)$(,$pat:pat => $res:expr)*}} => {
        {
            use $error::*;
            match $expr {
                $(std::result::Result::Err($return_error(err)) => return Err(err.into()),)*
                $($pat => $res,)*
            }
        }
    };
}
