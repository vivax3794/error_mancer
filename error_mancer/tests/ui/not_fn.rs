use error_mancer::prelude::*;

#[errors]
struct NotAFunction;

fn main() {}
