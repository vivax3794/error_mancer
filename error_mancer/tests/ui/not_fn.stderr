error: Expected function or impl block
 --> tests/ui/not_fn.rs:4:1
  |
4 | struct NotAFunction;
  | ^^^^^^^^^^^^^^^^^^^^
